dashboard/src/components/Table/PaginationInfo.tsx

```diff
@@ -33,28 +35,26 @@ interface IPaginationInfo {
     | Table<TreeTableBody>
     | Table<HardwareTableItem>
     | Table<Trees>;
-  data:
-    | TestByCommitHash[]
-    | TPathTests[]
-    | AccordionItemBuilds[]
-    | TreeTableBody[]
-    | HardwareTableItem[]
-    | Trees[];
   intlLabel: MessagesKey;
 }
 
 export function PaginationInfo({
   table,
-  data,
   intlLabel,
 }: IPaginationInfo): JSX.Element {
   const buttonsClassName = 'text-blue font-bold';
 
+  const filteredRowModel = table.getFilteredRowModel();
+  const countData = useMemo(
+    () => filteredRowModel.rows.length,
+    [filteredRowModel],
+  );
+
```

Comment:
you can skip filteredRowModel and get the countData _length_ directly, so you can also use this in the span below at line 67

Reply:
if I understood correctly, your suggestion is to do something like this

Reply:
If I understood correctly, your suggestion is to do something like this

`const countData = useMemo(
    () => table.getFilteredRowModel().rows.length,
    [table],
  );`

This way don't update the page count correctly

Reply:
As we discussed, it makes sense to do how you're doing, it is also done that way in other places. Also, my mistake, I confused countData with the old data.length, the way you're doing is right.